.github/workflows/build.yaml

```diff
@@ -47,11 +55,20 @@ jobs:
         with:
           node-version: 20
 
-      - name: Build and test
-        env:
-          DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
-          DOTNET_CLIE_TELEMETRY_OPTOUT: 1
+      - name: Setup non-windows build
+        if: matrix.os != 'windows-latest'
+        run: |
+          echo "DOTNET_CORE=1" >> $GITHUB_ENV
+      
+      - name: Install build dependencies
         run: |
           npm update -g npm --no-progress
           npm install --no-progress
-          npm test
+
+      - name: Build
```

Comment:
minor: build step is not required - `npm test` will build as a pre-requisite

Reply:
That's true, but it's easier to tell what failed when the phases are separated.